hpobench/benchmarks/mo/lm_benchmark.py

```diff
@@ -0,0 +1,368 @@
+        configuration
+        fidelity: Dict, None
+            epoch: int - Values: [1, 81]
+                Number of epochs an architecture was trained.
+                Note: the number of epoch is 1 indexed! (Results after the first epoch: epoch = 1)
+
+            Fidelity parameters, check get_fidelity_space(). Uses default (max) value if None.
+        rng : np.random.RandomState, int, None
+            Random seed to use in the benchmark.
+
+            To prevent overfitting on a single seed, it is possible to pass a
+            parameter ``rng`` as 'int' or 'np.random.RandomState' to this function.
+            If this parameter is not given, the default random state is used.
+
+
+        kwargs
+
+        Returns
+        -------
+        Dict -
+            function_value : Dict
+                validation_accuracy: float
+                log_perplexity: float
+            cost : time to train the network
+            info : Dict
+                validation_accuracy : float,
+                test_accuracy : float,
+                log_perplexity : float,
+                negative_log_perplexity : float,
+                training_cost : float,
+                valid_cost : float,
+                test_cost : float,
+                fidelity : Dict
+                    used fidelities in this evaluation
+        """
+
+        self.rng = rng_helper.get_rng(rng)
+        self.__seed_everything()
+
+        device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+        ts_start = time.time()
+
+        # batchify data
+        batch_size = configuration['batch_size']
+        train_data = batchify(self.X_train, batch_size=batch_size).to(device)
+        val_data = batchify(self.X_valid, batch_size=self.variable["eval_batch_size"]).to(device)
+        test_data = batchify(self.X_test, batch_size=self.variable["eval_batch_size"]).to(device)
+
+        epochs = fidelity['budget']
+
+        model = self.init_model(configuration).to(device)
+
+        criterion = nn.CrossEntropyLoss()
+
+        learning_rate = configuration['lr']
+        learning_rate_factor = configuration['lr_factor']
+        clip = configuration['clip']
+        best_val_loss = None
+        train_time = 0
+        eval_time = 0
+
+        t = tqdm.tqdm(total=epochs)
+        for epoch in range(epochs):
+            epoch_start_time = time.time()
+            train_loss, train_acc = model.train_fun(self.ntokens, criterion, train_data, learning_rate, clip)
+            train_time += time.time() - epoch_start_time
+            start = time.time()
+            val_loss, val_acc = model.eval_fun(self.ntokens, criterion, val_data)
+            val_loss = np.clip(val_loss, 1e-10, 10)
+            print("val acc for last epoch", val_acc)
+            eval_time += start - time.time()
+
+            t.set_postfix(val_accuracy=val_acc)
+            t.update()
+
+            if not np.isfinite(val_loss):
+                val_loss = 7
```

Comment:
is there a reason for the 7? 

Could you please add a comment to it, if that is from the authors